tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/jsf/NewLiferayJSFModuleProjectOp.java

```diff
@@ -0,0 +1,129 @@
+import org.eclipse.sapphire.PossibleValues;
+import org.eclipse.sapphire.Type;
+import org.eclipse.sapphire.Value;
+import org.eclipse.sapphire.ValueProperty;
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.ProgressMonitor;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.modeling.annotations.AbsolutePath;
+import org.eclipse.sapphire.modeling.annotations.DefaultValue;
+import org.eclipse.sapphire.modeling.annotations.DelegateImplementation;
+import org.eclipse.sapphire.modeling.annotations.Enablement;
+import org.eclipse.sapphire.modeling.annotations.FileSystemResourceType;
+import org.eclipse.sapphire.modeling.annotations.Label;
+import org.eclipse.sapphire.modeling.annotations.Listeners;
+import org.eclipse.sapphire.modeling.annotations.Required;
+import org.eclipse.sapphire.modeling.annotations.Service;
+import org.eclipse.sapphire.modeling.annotations.ValidFileSystemResourceType;
+
+/**
+ * @author Simon Jiang
+ */
+public interface NewLiferayJSFModuleProjectOp extends NewLiferayModuleProjectOp
+{
+
+    ElementType TYPE = new ElementType( NewLiferayJSFModuleProjectOp.class );
+
+    // *** ProjectName ***
+
+    @Label( standard = "project name" )
+    @Listeners( JSFModuleProjectNameListener.class )
+    @Service( impl = JSFModuleProjectNameValidationService.class )
+    @Required
+    ValueProperty PROP_PROJECT_NAME = new ValueProperty( TYPE, "ProjectName" );
+
+    Value<String> getProjectName();
+
+    void setProjectName( String value );
+
+    // *** ProjectLocation ***
+
+    @Type( base = Path.class )
+    @AbsolutePath
+    @Enablement( expr = "${ UseDefaultLocation == 'false' }" )
+    @ValidFileSystemResourceType( FileSystemResourceType.FOLDER )
+    @Label( standard = "location" )
+    @Service( impl = JSFModuleProjectLocationValidationService.class )
+    ValueProperty PROP_LOCATION = new ValueProperty( TYPE, "Location" );
+
+    Value<Path> getLocation();
+
+    void setLocation( String value );
+
+    void setLocation( Path value );
+
+    // *** UseDefaultLocation ***
+
+    @Type( base = Boolean.class )
+    @DefaultValue( text = "true" )
+    @Label( standard = "use default location" )
+    @Listeners( JSFModuleProjectUseDefaultLocationListener.class )
+    ValueProperty PROP_USE_DEFAULT_LOCATION = new ValueProperty( TYPE, "UseDefaultLocation" );
+
+    Value<Boolean> getUseDefaultLocation();
+
+    void setUseDefaultLocation( String value );
+
+    void setUseDefaultLocation( Boolean value );
+
+    @Service( impl = JSFModuleProjectArchetypeDefaultValueService.class )
+    ValueProperty PROP_ARCHETYPE = new ValueProperty( TYPE, "Archetype" );
+
+    Value<String> getArchetype();
+
+    void setArchetype( String value );
+
+    // *** JSF Project Template ***
+
+    @DefaultValue( text = "JSF Standard" )
+    @Label( standard = "Component Suite" )
+    @PossibleValues( values = { "JSF Standard", "Liferay Faces Alloy", "ICEFaces", "PrimeFaces", "RichFaces" } )
```

Comment:
We have this listed twice, once here and one in another place, we shouldn't repeat ourselves, s if we have to add another one, we have to do it in two places.